modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase

```diff
@@ -154,6 +154,342 @@ definition {
+				productName = "U-Joint",
+				productSku = "MIN55861",
+				southwestQuantity = "2");
+		}
+
+		task ("And minimum quantity is set to 8 and Allow Back Order is disabled") {
+			CommerceProducts.openProductsAdmin();
+			
+			CommerceNavigator.searchEntry(entryName = "U-Joint");
+
+			CommerceNavigator.gotoEntry(entryName = "U-Joint");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Configuration");
+			
+			CommerceProducts.updateProductInventoryConfiguration(
+				allowBackOrders = "false",
+				displayAvailability = "true",
+				displayStockQuantity = "true",
+				minOrderQuantity = "8",
+				publish = "true");
+		}
+
+		task ("And buyer user logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task("When user tries to add the order in the MiniCart"){
+			Click(
+			key_productName = "U-Joint",
+			locator1 = "CommerceAccelerators#ADD_TO_CART_BUTTON");
+		}
+
+		task ("Then the order item will not be add to the cart") {
+			Alert.viewErrorMessage(errorMessage = "Unable to add product to the cart.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to cart until there's stock quantity "
+	@priority = "3"
+	test CannotAddedMinimumQuantityUntilThereIsStock{
+		property portal.acceptance = "false";
+
+		task ("Given a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And given the warehouse quantity for the product is set to 6") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceEntry.editWarehousesQuantity(
+				italyQuantity = "2",
+				northeastQuantity = "2",
+				productName = "U-Joint",
+				productSku = "MIN55861",
+				southwestQuantity = "2");
+		}
+
+		task ("And minimum quantity is set to 4 and Allow Back Order is disabled") {
+			CommerceProducts.openProductsAdmin();
+			
+			CommerceNavigator.searchEntry(entryName = "U-Joint");
+
+			CommerceNavigator.gotoEntry(entryName = "U-Joint");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Configuration");
+			
+			CommerceProducts.updateProductInventoryConfiguration(
+				allowBackOrders = "false",
+				displayAvailability = "true",
+				displayStockQuantity = "true",
+				minOrderQuantity = "4",
+				publish = "true");
+		}
+
+		task ("And buyer user logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("And buyer user can do the checkout with the quantity that there's in stock quantity "){
```

Comment:
task ("And buyer user can do the checkout with a quantity that is less than the stock quantity one") {